src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1488,6 +1490,28 @@ private void ExecuteRestartSearchCommand(object sender, ExecutedRoutedEventArgs
                 .PostOnFailure(nameof(PackageManagerControl), nameof(ExecuteRestartSearchCommand));
         }
 
+        private void ExecuteSearchPackageCommand(object sender, ExecutedRoutedEventArgs e)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            var alternatePackageId = e.Parameter as string;
+            if (!string.IsNullOrWhiteSpace(alternatePackageId))
+            {
+                if (_windowSearchHost != null && _windowSearchHost.IsEnabled)
+                {
+                    if (_windowSearchHost.SearchTask != null)
+                    {
+                        // Terminate current search by reporting it as complete
+                        _currentSearchCallback?.ReportComplete(_windowSearchHost.SearchTask, dwResultsFound: 0);
+                    }
```

Comment:
Could this block change the state of _windowSearchHost? If not, the next conditional if block should be an else. If this could change the _windowSearchHost state, from not null to null, then the next conditional should remain a separate check.

Reply:
`ReportComplete()` changes the state of `_windowSearchHost`; among other things, it sets `_windowSearchHost.SearchTask` to `null`; which is needed to create a new search.

Reply:
Sounds good, thanks! I'm going to resolve this comment thread.